features/step_definitions/api-token.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+const Assert = require('chai').assert;
+const jwt = require('jsonwebtoken');
+const request = require('../support/request');
+const testTokenName = 'an access token';
+
+module.exports = function server() {
+    this.When(/^a new API token is generated$/, () =>
+        // Make sure there won't be a conflict: delete the token if it's already there
+        request({
+            uri: `${this.instance}/${this.namespace}/tokens`,
+            method: 'GET',
+            auth: {
+                bearer: this.jwt
+            }
+        }).then((response) => {
+            if (response.body === '[]') {
+                return 0;
+            }
+
+            console.log(response.body);
+
+            const id = JSON.parse(response.body)
+                .find(token => token.name === testTokenName)
+                .id;
+
+            return request({
+                uri: `${this.instance}/${this.namespace}/tokens/${id}`,
+                method: 'DELETE',
+                auth: {
+                    bearer: this.jwt
+                }
+            });
+        }).then(() =>
+            request({
+                uri: `${this.instance}/${this.namespace}/tokens`,
+                method: 'POST',
+                auth: {
+                    bearer: this.jwt
+                },
+                body: {
+                    name: testTokenName
+                },
+                json: true
+            })).then((response) => {
+                Assert.equal(response.statusCode, 201);
+                this.userAccessToken = response.body.value;
```

Comment:
Should also assert that the body has something in `value`. Otherwise, it'll fail in the next step and the cause of the error won't be deterministic